nemo/collections/asr/metrics/der.py

```diff
@@ -32,6 +34,83 @@
 ]
 
 
+def get_partial_ref_labels(pred_labels: List[str], ref_labels: List[str]) -> List[str]:
+    """
+    For evaluation of online diarization performance, generate partial reference labels 
+    from the last prediction time.
+
+    Args:
+        pred_labels (list[str]): list of partial prediction labels
+        ref_labels (list[str]): list of full reference labels 
+
+    Returns:
+        ref_labels_out (list[str]): list of partial reference labels
+    """
+    # If there is no reference, return empty list
+    if len(ref_labels) == 0:
+        return []
+
+    # If there is no prediction, set the last prediction time to 0
+    if len(pred_labels) == 0:
+        last_pred_time = 0
+    else:
+        # The lastest prediction time in the prediction labels
+        last_pred_time = max([float(labels.split()[1]) for labels in pred_labels])
+    ref_labels_out = []
+    for label in ref_labels:
+        start, end, speaker = label.split()
+        start, end = float(start), float(end)
+        # If the current [start, end] interval is latching the last prediction time
```

Comment:
latching -> matching

Reply:
Changed the expression (Checked by Elena)